numpy/ma/core.py

```diff
@@ -7887,19 +7887,10 @@ def _pickle_warn(method):
 
 
 def dump(a, F):
-    """
-    Pickle a masked array to a file.
-
-    This is a wrapper around ``cPickle.dump``.
-
-    Parameters
-    ----------
-    a : MaskedArray
-        The array to be pickled.
-    F : str or file-like object
-        The file to pickle `a` to. If a string, the full path to the file.
-
-    """
+    # NumPy 1.15.0, 2017-12-10
+    warnings.warn(
```

Comment:
There is the `_pickle_warn` function, which was handling the current DeprecationWarning, I think you may have missed it.

Reply:
So, I should remove `_pickle_warn` or `warnings.warn`?
I introduced the latter since this was consistent in all the deprecated functions I've seen so far.

Reply:
The warning was already in place since a long time. You can just remove these functions entirely I think. After your change, there would be two warnings given.